aggregator/src/aggregator.rs

```diff
@@ -1904,6 +1910,30 @@ impl VdafOps {
                         .add(1, &[KeyValue::new("type", "duplicate_extension")]);
                     return Err(PrepareError::InvalidMessage);
                 }
+
+                if require_taskprov_extension {
+                    if !extensions.get(&ExtensionType::Taskprov).map(|data| data.is_empty()).unwrap_or(false) {
+                        debug!(
+                            task_id = %task.id(),
+                            metadata = ?prepare_init.report_share().metadata(),
+                            "Taskprov task received report with missing or malformed taskprov extension",
+                        );
+                        aggregate_step_failure_counter
+                            .add(1, &[KeyValue::new("type", "missing_or_malformed_taskprov_extension")]);
+                        return Err(PrepareError::InvalidMessage);
+                    }
+                } else if extensions.contains_key(&ExtensionType::Taskprov) {
+                    // taskprov not enabled, but the taskprov extension is present.
+                        debug!(
+                            task_id = %task.id(),
+                            metadata = ?prepare_init.report_share().metadata(),
+                            "Taskprov task received report with unexpected taskprov extension",
```

Comment:
This message is incorrect
```suggestion
                            "Non-Taskprov task received report with unexpected taskprov extension",
```